packages/material-ui/src/Tooltip/Tooltip.test.js

```diff
@@ -24,15 +24,19 @@ function simulatePointerDevice() {
 }
 
 describe('<Tooltip />', () => {
+  /**
+   * @type {ReturnType<typeof useFakeTimers>}
+   */
   let clock;
   beforeEach(() => {
     testReset();
     clock = useFakeTimers();
   });
 
   afterEach(() => {
-    clock.tick(800); // cleanup the hystersis timer
```

Comment:
This is meant for 
https://github.com/mui-org/material-ui/blob/042ccfef20688ecb12fccefe74a27e59a83bbc5f/packages/material-ui/src/Tooltip/Tooltip.js#L328

It was added to get tests fails before working on a fix. I think that we should keep the logic to start each test from a pristine state.

Reply:
>  I think that we should keep the logic to start each test from a pristine state.

To be clear: `restore()` in afterEach and `testReset` means that everything is pristine. If this was supposed to test something then it should be included as an explicit assertion. Test hooks are not well suited for test assertions.

Reply:
Cool, I haven't realized that restore() would achieve the same outcome. Yeah, you are probably right, I should/could have written a test with a nested after each call to make sure the cleanup was correct.

Reply:
restore() removes all timers. If you want to make sure running the clock for 800ms doesn't crash or warn or make another assertion then it should be handled in the test itself. I'll go back to the original PR to see what the issue was. I suspect we didn't restore the clock between each test but only after the full suite. In that scenario it makes sense to let timers run between tests. But this is still brittle because you have to hardcode the timeout.

Reply:
Should be fine if https://github.com/mui-org/material-ui/pull/21805 is green and Josh agrees.

Reply:
Build green and Josh agrees: https://github.com/mui-org/material-ui/pull/21805#pullrequestreview-449044258

Reply:
In case of potential confusion, if you blame this line, you will find @oliviertassinari as the author 40b2a8b.